### ISSUE TITLE
ESP32 code does not compile

### ISSUE THREAD

**Issue by simowilso** [projectswithalex/Reaction-Lights-Training-Module#4]
Hi - thanks for sharing your code.
I am not a programmer but trying to build some of these for my wife who is recovering from a stroke.

I am intending to use esp8266, but, I also tried an esp32 board while I'm waiting for parts to arrive. Wen compiling I get this error:

C:\Users\siwil\AppData\Local\Temp\.arduinoIDE-unsaved2024113-30004-1ckbaul.05eo\sketch_dec3b\sketch_dec3b.ino: In function 'void initESPNOWcomm()':
C:\Users\siwil\AppData\Local\Temp\.arduinoIDE-unsaved2024113-30004-1ckbaul.05eo\sketch_dec3b\sketch_dec3b.ino:120:50: error: invalid conversion from 'void (*)(const uint8_t*, const uint8_t*, int)' {aka 'void (*)(const unsigned char*, const unsigned char*, int)'} to 'esp_now_recv_cb_t' {aka 'void (*)(const esp_now_recv_info*, const unsigned char*, int)'} [-fpermissive]
  120 |   esp_err_t startsend = esp_now_register_recv_cb(dataReceived);          // this function will get called whenever we receive data
      |                                                  ^~~~~~~~~~~~
      |                                                  |
      |                                                  void (*)(const uint8_t*, const uint8_t*, int) {aka void (*)(const unsigned char*, const unsigned char*, int)}
In file included from C:\Users\siwil\AppData\Local\Temp\.arduinoIDE-unsaved2024113-30004-1ckbaul.05eo\sketch_dec3b\sketch_dec3b.ino:25:
C:\Users\siwil\AppData\Local\Arduino15\packages\esp32\tools\esp32-arduino-libs\idf-release_v5.3-a0f798cf\esp32c3/include/esp_wifi/include/esp_now.h:158:54: note:   initializing argument 1 of 'esp_err_t esp_now_register_recv_cb(esp_now_recv_cb_t)'
  158 | esp_err_t esp_now_register_recv_cb(esp_now_recv_cb_t cb);
      |                                    ~~~~~~~~~~~~~~~~~~^~

exit status 1

Compilation error: invalid conversion from 'void (*)(const uint8_t*, const uint8_t*, int)' {aka 'void (*)(const unsigned char*, const unsigned char*, int)'} to 'esp_now_recv_cb_t' {aka 'void (*)(const esp_now_recv_info*, const unsigned char*, int)'} [-fpermissive]

**Comment by TheprofessionalKhaled** [comment]
hi donnot try it also on esp8266 ...for more than  a whole year literally ..I try to operate that project...I spend alots of money....ofcourse you want to ask why I still determined ....because I am electrical engineer still junior also I studied programming ....so I take it as a target or challenge for learning purpose nothing more ...even if my beginning intention is to train but now it is for learning ......so if you arenot will then donnot start or spend money

**Comment by TheprofessionalKhaled** [comment]
hi how are you? does the project works fine with you after your last update

**Comment by simowilso** [comment]
I have it working with three lights so far, it took some time to finish all the 3d printing and for the battery hardware etc. I haven't hooked up the battery voltage sensors yet. Today I discovered that the range sensors do not work in bright light, I don't know yet if that can be improved.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by simowilso** [comment]
https://youtube.com/shorts/3FEB_nFHJvw?si=m7mhqE5AyYohv1dL

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by TheprofessionalKhaled** [comment]
For me it still don’t work even after the update
Sometimes when I push the button the led color changes sometimes it doesn’t
work that color freezes on red with no respose and other time the color
changes but the sensor doesn’t read the signal or interrupt
I think it still be apower issue may be I need to add capacitors
Also i think if i feed the sensor from external 3v supply and make the
whole project with common ground that will fix the issue

**Comment by TheprofessionalKhaled** [comment]
congratulation first
so please tell me if I use the same code like you ...changes the values like you and I am sure that the problem not from the mac address ....what is the difference betweeen yours and mine that makes your project works

is that from the power circuit ...do you use a specific battery or charger with certian specification ? what do you think ?

**Comment by simowilso** [comment]
You can see all the changes I have made here https://github.com/simowilso/Reaction-Lights-Training-Module

It is still not 100%, sometimes on startup the master ECU (ecu1) doesn't initialise the TOF sensor correctly - I'm still investigating that.
________________________________

**Comment by TheprofessionalKhaled** [comment]
How are you man any news , do you get why Sensor stopped functioning ?
All what I want to try is powering the sensor separately from another
source and make all the project with common ground

**Comment by simowilso** [comment]
I committed a change today to ecu1.ino trying to get the sensor 100% reliable, I don't know if it's successful yet.

**Comment by TheprofessionalKhaled** [comment]
I am really enthusiastic to try it after returning from traveling at
Thursday

**Comment by TheprofessionalKhaled** [comment]
Hi man do you reach any new solutions or find any new points about the
issue ?

**Comment by simowilso** [comment]
Nope, haven't spent a huge amount of time on it. The correct code is being sent to the time of flight sensor,  but for some reason some times it doesn't initialise.  When it works it's great. I think the next step is to try one of the other ToF libraries.

**Comment by TheprofessionalKhaled** [comment]
You know after serial print I found (non master ) didn’t recieve packets I
think espnow maybe the issue especially I found it has limited
functionality for esp8266 and its full functionality for esp32 i think I
should migrate to esp32 so what do you think
